### PR TITLE
feat: New Module `avm/res/insights/webtest`

### PR DESCRIPTION
**Pull request by krbar**
## Description

New Module `avm/res/insights/webtest`, migrated from CARML.

## Adding a new module

- [x] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Pipeline references

| Pipeline |
| - |
| [avm.res.insights.webtest](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.insights.webtest.yml) |

### CODE REVIEW

**Comment by krbar** [comment]
@AlexanderSehr This module doesn't pass the static validation because the parameter `tags` is required and therefore cannot be nullable. Any suggestion how to deal with this?

**Comment by ChrisSidebotham** [comment]
@krbar - Could it be worth adding a parameter for the required hidden-link tag and concatenate this with the nullable tag parameter required by the static validation tests?

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -63,7 +63,7 @@
 /avm/res/insights/metric-alert/ @Azure/avm-res-insights-metricalert-module-owners-bicep @Azure/avm-core-team-technical-bicep
 #/avm/res/insights/private-link-scope/ @Azure/avm-res-insights-privatelinkscope-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/insights/scheduled-query-rule/ @Azure/avm-res-insights-scheduledqueryrule-module-owners-bicep @Azure/avm-core-team-technical-bicep
-#/avm/res/insights/webtest/ @Azure/avm-res-insights-webtest-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/insights/webtest/ @Azure/avm-res-insights-webtest-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (ChrisSidebotham) [inline .github/CODEOWNERS:66]: Please ensure you have created the required teams. Currently unable to locate this team

File: avm/res/insights/webtest/main.bicep [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+metadata owner = 'Azure/module-maintainers'
+
+@sys.description('Required. Name of the webtest.')
+param name string
+
+@sys.description('Required. User defined name if this WebTest.')
+param webTestName string
+
+@sys.description('Required. A single hidden-link tag pointing to an existing AI component is required.')
+param tags object
```
Comment (ChrisSidebotham) [inline avm/res/insights/webtest/main.bicep:12]: Outstanding decision on how to action this with this resource type

**Review by ChrisSidebotham — DISMISSED** [review]

File: avm/res/insights/webtest/main.bicep (hunk 2 above)
Comment (AlexanderSehr) [reply avm/res/insights/webtest/main.bicep:12]: Your suggest is actually a pretty neat one. If the RT needs one specific tag of a certain type - then we can just have a mandatory parameter for that. @krbar what's your take on that?

Comment (krbar) [reply avm/res/insights/webtest/main.bicep:12]: @ChrisSidebotham thanks for the suggestion, we can do it like this. The mandatory tag contains the `resourceId` of an Application Insights resource, so would create a mandatory parameter `appInsightResourceId`. 
However I suggest to add a short note to the description of the `tags` parameter informing the user about an additional tag.

CC: @AlexanderSehr 

This is the mandatory tag:
`'hidden-link:${appInsightResourceId}': 'Resource'`

File: .github/CODEOWNERS (hunk 1 above)
Comment (krbar) [reply .github/CODEOWNERS:66]: Teams area created

File: avm/res/insights/webtest/main.bicep (hunk 2 above)
Comment (krbar) [reply avm/res/insights/webtest/main.bicep]: Like this? [541822f](https://github.com/Azure/bicep-registry-modules/pull/728/commits/541822f049fb695f1a28007325e85258900167d2)

File: avm/res/insights/webtest/main.bicep [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+@sys.description('Required. Name of the webtest.')
+param name string
+
+@sys.description('Required. Resource ID of the App Insights resource to link with this webtest.')
+param appInsightResourceId string
+
+@sys.description('Required. User defined name if this WebTest.')
+param webTestName string
+
+@sys.description('Optional. Resource tags. Note: a mandatory tag \'hidden-link\' will be automatically added to the tags defined here.')
```
Comment (AlexanderSehr) [inline avm/res/insights/webtest/main.bicep:14]: Sounds good. The only thing is add is a reference to the parameter the value is taken from, i.e., the resource Id

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]

File: avm/res/insights/webtest/tests/e2e/max/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    location: location
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
```
Comment (AlexanderSehr) [inline avm/res/insights/webtest/tests/e2e/max/main.test.bicep:49]: Role Assignment Test missing

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**